### PR TITLE
rest api: separate example to individual pages and change config

### PR DESCRIPTION
**Pull request by ykethan**
#### Related GitHub issue #, if available:
closes: https://github.com/aws-amplify/docs/issues/7487
closes: https://github.com/aws-amplify/docs/issues/7486
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+export function getStaticProps(context) {
+  return {
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+To setup and configure your application with HTTP API to make requests to [Amazon API Gateway](https://aws.amazon.com/api-gateway/) and trigger the lambda function, you can create a HTTP API using the AWS Cloud Development Kit (AWS CDK).
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx:29]:
```suggestion
Using the [AWS Cloud Development Kit (AWS CDK)](https://aws.amazon.com/cdk/), you can configure Amplify Functions as resolvers for routes of an [HTTP API powered by Amazon API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api.html).
```

File: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+## Set up HTTP API with Lambda Function
+
+Create a new directory and a resource file, `amplify/functions/api-function/resource.ts`. Then, define the function with `defineFunction`:
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx:33]:
```suggestion
To get started, create a new directory and a resource file, `amplify/functions/api-function/resource.ts`. Then, define the function with `defineFunction`:
```

File: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+```ts title="amplify/functions/api-function/resource.ts"
+import { defineFunction } from "@aws-amplify/backend";
+
+export const myApiFunction = defineFunction({
+  name: "api-function",
+});
+```
+
+create the corresponding handler file, `amplify/functions/api-function/handler.ts`, file with the following contents:
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx:43]:
```suggestion
Then, create the corresponding handler file, `amplify/functions/api-function/handler.ts`, file with the following contents:
```

File: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+export const handler: APIGatewayProxyHandlerV2 = async (event) => {
+  console.log("event", event);
+  return {
+    statusCode: 200,
+    // Modify the CORS settings below to match your specific requirements
+    headers: {
+      "Access-Control-Allow-Origin": "*", // Restrict this to domains you trust
+      "Access-Control-Allow-Headers": "*", // Specify only the headers you need to allow
+    },
+    body: JSON.stringify("Hello from myFunction!"),
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx:57]:
```suggestion
    body: JSON.stringify("Hello from api-function!"),
```

File: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+  };
+};
+```
+
+Use the [AWS CDK](https://docs.aws.amazon.com/cdk/latest/guide/home.html) to create an HTTP API resource powered by [Amazon API Gateway](https://aws.amazon.com/api-gateway/).
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx:62]:
```suggestion
Next, using the AWS CDK, create an HTTP API in your backend file:
```

File: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+  ...existingConfig,
+  API: {
+    REST: outputs.custom.API,
+  },
+});
+```
+</InlineFilter>
+
+<InlineFilter filters={["nextjs"]}>
+```tsx title="pages/index.tsx"
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx:237]:
````suggestion
```tsx title="pages/_app.tsx"
````

this is ideally at the app level in next.js

File: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+      },
+    },
+  },
+});
+```
+
+## Install Amplify Libraries
+
+<InlineFilter filters={['javascript', "angular", "react", "vue", "react-native", "nextjs"]}>
+Use the package manager of your choice to install the amplify JS library. For example, with `npm`:
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx:186]:
```suggestion
Use the package manager of your choice to install the Amplify JavaScript library. For example, with `npm`:
```

consolidating the reference to the js library (could also call this "client library" or "Amplify client library"

File: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+
+```bash title="Terminal" showLineNumbers={false}
+npm add aws-amplify
+```
+</InlineFilter>
+
+<InlineFilter filters={["react-native"]}>
+Use the package manager of your choice to install the amplify JS library. For example, with `npm`:
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx:194]:
```suggestion
Use the package manager of your choice to install the Amplify JavaScript library. For example, with `npm`:
```

File: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+
+## Initialize Amplify API
+
+<InlineFilter filters={['javascript', "angular", "react", "vue", "react-native", "nextjs"]}>
+To initialize the Amplify API category you need to configure Amplify with `Amplify.configure()`.
+
+Import and load the configuration file in your app. It's recommended you add the Amplify configuration step to your app's root entry point. For example `index.js` in React or `main.ts` in Angular.
+
+<InlineFilter filters={['javascript', "angular", "react", "vue", "react-native"]}>
+```javascript title="src/index.js"
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-http-api/index.mdx:221]:
````suggestion
```ts title="src/main.ts"
````

File: src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-rest-api/index.mdx [hunk 10]
```diff
@@ -26,7 +26,7 @@ export function getStaticProps(context) {
   };
 }
 
-To setup and configure your application with REST API or HTTP API to make requests to [Amazon API Gateway](https://aws.amazon.com/api-gateway/) and trigger the lambda function, you can create a REST or HTTP API using the AWS Cloud Development Kit (AWS CDK).
+To setup and configure your application with REST API to make requests to [Amazon API Gateway](https://aws.amazon.com/api-gateway/) and trigger the lambda function, you can create a REST using the AWS Cloud Development Kit (AWS CDK).
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/add-aws-services/rest-api/set-up-rest-api/index.mdx:29]: can we do a similar rewording to the intro here?

**Review by josefaidt — APPROVED** [review]

**Review by chrisbonifacio — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Pull request merged**